### ISSUE TITLE
infinite error messages

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11305]
|<img alt="maoy's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| maoy reported|
|-|-|
|Trac ID|trac#374|
|Type|defect|
|Created|2003-10-29 02:08:14Z|

<details><summary>Searchable metadata</summary>

```
trac-id__374 374
type__defect defect
reporter__maoy maoy
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__invalid invalid
component__core core
keywords__ 
time__1067393294000000 1067393294000000
changetime__1469075050051204 1469075050051204
version__ 
owner__ 
cc__itamarst cc__jknight cc__maoy
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="maoy's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|maoy commented|
|-|-|

```
#!html
<pre>
The infinite error messages(no fileno method) happens when the connections has
been closed, but the startReading() or other related functions are called
afterwards. This usually happens when producers and consumers are involved. I
admit that if the twisted programmer write the program very carefully, it should
have been avoided. However, in case something bad happened, the infinite err
msgs are still very annoying. It could fill in your log disk quickly. I would
suggest add a flag or something to prevent people from calling startReading()
after the connection has been closed. Thanks.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Is it possible to have code which shows this bug?
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="maoy's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|maoy commented|
|-|-|

```
#!html
<pre>
OK. How about this one

from twisted.internet.protocol import Protocol, ClientFactory
from sys import stdout

class Echo(Protocol):

    def dataReceived(self, data):
        stdout.write(data)
        self.transport.loseConnection()

    def connectionLost(self):
        self.transport.startReading()

class EchoClientFactory(ClientFactory):

    def buildProtocol(self, addr):
        print 'Connected.'
        return Echo()

from twisted.internet import reactor
reactor.connectTCP('158.130.68.33', 22, EchoClientFactory())
reactor.run()

The code looks really stupid. But in reality, it might happen in more 
subtle ways. E.g, when you have producer/consumers.

Anyway, I don't think infinite err msg is a good idea. ;) Hope this helps.

Yun

On Fri, 7 Nov 2003, Moshe Zadka [Twisted issue tracker] wrote:

> 
> Moshe Zadka &lt;moshez@...> added the comment:
> 
> Is it possible to have code which shows this bug?
> 
> ----------
> status: unread -> chatting
> ____________________________________________________________________
> Twisted issue tracker &lt;twisted.roundup@...>
> &lt;http://www.twistedmatrix.com/users/roundup.twistd/twisted/#374>
> ____________________________________________________________________
>
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
I get no infinite error messages with provided test, so perhaps this has 
been fixed already. That testcase (with a local server, of course) should 
be added, though.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

I think this is related to [#11072](https://github.com/twisted/twisted/issues/11072)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

After fixing a couple problems with the example, I also don't see an infinite stream of errors - just one.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|GitHub commented|
|-|-|

In changeset 6a4bbc8a7750e7f2e5a2c2c95347203bdf9a5059
```
#!CommitTicketReference repository="" revision="6a4bbc8a7750e7f2e5a2c2c95347203bdf9a5059"
Merge pull request #374 from twisted/8640-scop-release-stdout

Author: scop
Reviewer: markrwilliams, rodrigc
Fixes: #8640

Close runCommand stdout using communicate
```
